libraries/AP_Mount/AP_Mount_SToRM32_serial.cpp

```diff
@@ -59,7 +59,8 @@ void AP_Mount_SToRM32_serial::update()
 
         // point to the angles given by a mavlink message
         case MAV_MOUNT_MODE_MAVLINK_TARGETING:
-            // do nothing because earth-frame angle targets (i.e. _angle_ef_target_rad) should have already been set by a MOUNT_CONTROL message from GCS
+            // This is not a MAVLINK-based protocol, so we need to send the coordinates ourselves
```

Comment:
We should leave the old message in. In any case, I don't really understand the comment about this not being a MAVLink-based protocol, what does that have to do with it?

Reply:
Thanks for the review!

My understanding is that when the mount is in MAV_MOUNT_MODE_MAVLINK_TARGETING, and the mount is set to a MAVLINK protocol, the MOUNT_CONTROL messages are forwarded to the mount elsewhere, according to the MAVLINK routing rules. That's how I understood this comment, as it seems to be copied from AP_Mount_SToRM32.cpp, where it does make sense. However, the SToRM32 Serial protocol is not MAVLINK based, therefore the same logic does not apply. This is why I changed the comment, and also why I didn't touch AP_Mount_SToRM32.cpp. To be honest, I haven't thoroughly checked if the behavior was correct in STorM32 MAVLINK mode.

Please let me know if I got something wrong.

Reply:
> We should leave the old message in. In any case, I don't really understand the comment about this not being a MAVLink-based protocol, what does that have to do with it?

Oh, sorry, after reading the old message 10 times I got it's meaning :) I did think that "doing nothing" referred to sending the coordinates to the gimbal.

Reply:
> My understanding is that when the mount is in MAV_MOUNT_MODE_MAVLINK_TARGETING, and the mount's serial port is set to a MAVLINK protocol, the MOUNT_CONTROL messages are forwarded to the mount elsewhere, according to the MAVLINK routing rules.

That would only happens if you the target system or component of the message is set to broadcast or to the IDs of the gimbal (in which case you wouldn't be using the ArduPilot driver at all) - I don't know if the STorM32 accepts a message not coming from component 1 (autopilot) though.

> Oh, sorry, after reading the old message 10 times I got it's meaning :) I did think that "doing nothing" referred to sending the coordinates to the gimbal.

🙂 Yes, it doesn't refer to that, it's just that setting the angles in the variables are done when receiving the MAVLink message and not when the update method is run (as happens for the other modes). I do understand your confusion though, I still remember that I also didn't get its meaning the first I read it.

> Should I still update AP_Mount_SToRM32.cpp[...]?

Yes, I would think so, I don't see any reason to not do it. As explained above, MAVLink routing might send the message to the gimbal if it broadcasts, but it's not necessarily done that way (in my opinion it makes more sense to direct the message at the autopilot).

Reply:
Done!